overlord/devicestate/devicemgr.go

```diff
@@ -818,6 +818,14 @@ var ErrUnsupportedAction = errors.New("unsupported action")
 // system reboot will be requested when the request can be successfully carried
 // out.
 func (m *DeviceManager) RequestSystemAction(systemLabel string, action SystemAction) error {
+	if systemLabel == "" {
+		return fmt.Errorf("internal error: system label is unset")
+	}
+	if m.systemMode == "" {
```

Comment:
mmh, we don't do this anywhere else, usually we use the grade unset of the model to know if we are on uc16/18. we need to think a bit.